cedar-policy/src/api.rs

```diff
@@ -1720,6 +1751,47 @@ impl PolicySet {
         Ok(())
     }
 
+    /// Remove a `Template` from the `PolicySet`.
+    /// If any policy is linked to the template, this will error
+    pub fn remove_template(&mut self, template_id: PolicyId) -> Result<Template, PolicySetError> {
+        let Some(template) = self.templates.remove(&template_id) else {
+            return Err(PolicySetError::TemplateNonexistentError(template_id));
+        };
+        // If self.templates and self.ast disagree, authorization cannot be trusted.
+        // PANIC SAFETY: We just found the policy in self.templates.
+        #[allow(clippy::panic)]
+        match self
+            .ast
+            .remove_template(&ast::PolicyID::from_string(template_id.to_string()))
+        {
+            Ok(_) => Ok(template),
+            Err(ast::PolicySetTemplateRemovalError::RemoveTemplateWithLinksError(_)) => {
+                self.templates.insert(template_id.clone(), template);
+                Err(PolicySetError::RemoveTemplateWithActiveLinksError(
+                    template_id,
+                ))
+            }
+            Err(ast::PolicySetTemplateRemovalError::RemovePolicyNoTemplateError(_)) => {
+                panic!("Found template policy in self.templates but not in self.ast");
+            }
+        }
+    }
+
+    /// Get policies linked to a `Template` in the `PolicySet`.
+    /// If any policy is linked to the template, this will error
+    pub fn get_linked_policies(
+        &mut self,
+        template_id: PolicyId,
+    ) -> Result<Vec<PolicyId>, PolicySetError> {
```

Comment:
same here we could return `Iterator<Item = PolicyId>` to avoid the `collect()`

Reply:
Doing this above is fine, but here we convert from `PolicyID` to `PolicyId` so I think we'd need to introduce a new type `PolicyIdRef(&PolicyID)` to handle this, which doesn't seem worth it to avoid the `.collect()` to me.

Reply:
That is fine

Reply:
Or, (not exactly seeing your code) does it work to use `.ref_cast()` to turn `&PolicyID` into `&PolicyId`?

Reply:
(as we do for other types many other places in api.rs)

Reply:
Worked like a charm